### ISSUE TITLE
Mapper Bean not found when adding new entity with a .jh file. 

### ISSUE THREAD

**Issue by catalindumitrache76** [jhipster/generator-jhipster#6170]
##### **Overview of the issue**

I created a new monolythic app, and when I tried adding a new entity with a productos_fitosanitarios.jh file and the command "jhipster import-jdl productos_fitosanitarios.jh" it would import all the changes (I selected the option "a, override all files") but the app would not start due to the following error message: 

2017-07-25 19:12:35.537  WARN 14472 --- [  restartedMain] ationConfigEmbeddedWebApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'fitosanitarioServiceImpl' defined in file [/home/catalin/Escritorio/TFG/TFG/build/classes/java/main/com/mycompany/myapp/service/impl/FitosanitarioServiceImpl.class]: Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'com.mycompany.myapp.service.mapper.FitosanitarioMapper' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
2017-07-25 19:12:35.660 ERROR 14472 --- [  restartedMain] o.s.b.d.LoggingFailureAnalysisReporter   : 

***************************
APPLICATION FAILED TO START
***************************

Description:

Parameter 1 of constructor in com.mycompany.myapp.service.impl.FitosanitarioServiceImpl required a bean of type 'com.mycompany.myapp.service.mapper.FitosanitarioMapper' that could not be found.

Action:

Consider defining a bean of type 'com.mycompany.myapp.service.mapper.FitosanitarioMapper' in your configuration.

Process finished with exit code 0  

##### **Motivation for or Use Case**

The app won't start. 

##### **Reproduce the error**

I created an app from scratch, and just tried to add an entity through a .jh file and using the "jhipster import-jdl productos_fitosanitarios.jh" command. 

##### **Related  issues**

Yes, but it was not answered: https://github.com/jhipster/generator-jhipster/issues/5464

##### **Suggest a Fix**

I think the problem is that a bean annotation is not created among the 
com.mycompany.myapp.service.mapper.FitosanitarioMapper class

##### **JHipster Version(s)**

```
tfg-j-hipster-app@0.0.0 /home/catalin/Escritorio/TFG/TFG
└── generator-jhipster@4.6.1 

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**
<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp"
    },
    "jhipsterVersion": "4.6.1",
    "baseName": "TfgJHipsterApp",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "session",
    "hibernateCache": "no",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "gradle",
    "enableSocialSignIn": false,
    "rememberMeKey": "replaced-by-jhipster-info",
    "clientFramework": "angular1",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": false
  }
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Fitosanitario (fitosanitario) {
  numregistro String required,
  nombrecomercial String,
  titular String,
  formulado String
}
dto Fitosanitario with mapstruct
paginate Fitosanitario with pagination
angularSuffix Fitosanitario with mySuffix

</pre>
</details>

##### **Environment and Tools**
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)
git version 2.7.4
node: v7.9.0
npm: 5.3.0
bower: 1.8.0
gulp:
[19:35:50] CLI version 1.3.0
[19:35:50] Local version 3.9.1
yeoman: 2.0.0
yarn: 0.24.5
Docker version 1.12.6, build 78d1802

##### **Browsers and Operating System**

I am using Ubuntu 16.4 and IntelliJ IDEA 2017.1.3. 

- [ X ] Checking this box is mandatory (this is just to show you read everything)

**Comment by jdubois** [comment]
Use the search engine, this has been answered many times

**Comment by atomfrede** [comment]
Do you start the app via cli or via idea?

Am 25.07.2017 7:52 nachm. schrieb "Catalin Dumitrache" <
notifications@github.com>:

**Comment by catalindumitrache76** [comment]
@jdubois You were right, the problem was that I thought Idea had the “Compile on save” feature enabled. Sorry, my bad. Actually, @felipebcs on this comment gave me the clue: "If anyone land on this page looking for a possible solution:
I had a similar problem using Maven and Eclipse. At the end I realized that I hadn't read well the bottom of the "Configuring your IDE" page, which says about MapStruct, but I also had to add the target/generated-sources folder to my build path (as instructed on http://g00glen00b.be/mapstruct/ )."
